SD_Assignment1/src/main/java/ro/utcn/aron/persistence/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,409 @@
+package ro.utcn.aron.persistence.jdbc;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+import ro.utcn.aron.model.Answer;
+import ro.utcn.aron.model.Question;
+import ro.utcn.aron.model.Vote;
+import ro.utcn.aron.persistence.api.QuestionRepository;
+
+
+public class JdbcQuestionRepository implements QuestionRepository {
+
+	private  final JdbcTemplate template;
+
+	public JdbcQuestionRepository(JdbcTemplate template) {
+		this.template = template;
+	}
+
+	@Override
+	public Question save(Question question) {
+		if (question.getId() != 0) {
+			update(question);
+		} else {
+			int id = insert(question);
+			question.setId(id);
+		}
+
+		return question;
+	}
+
+	@Override
+	public Optional<Question> findById(int id) {
+		List<Question> questions = template.query("SELECT * FROM question WHERE id = ?", new Object[] { id },
+				(resultSet, i) -> new Question(resultSet.getInt("id"), resultSet.getString("title"),
```

Comment:
You reuse this RowMapper lambda between this method and the findAll method --> might be an idea to extract it to its own class.